packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -149,6 +150,22 @@ class AssembleCommand extends FlutterCommand {
     return const CustomDimensions();
   }
 
+  @override
+  Future<Event> unifiedAnalyticsUsageValues(String commandPath) async {
+    final FlutterProject flutterProject = FlutterProject.current();
```

Comment:
From a brief look at `FlutterProject.current()`, it looks like that does a non-trivial amount of work. Also, it looks like we're duplicating that work three times in this class. Can we lift it to a `late final` field on the class, and re-use the work?

Reply:
Yeah good call, we were actually calling it several times in that file, this should help speed things up